src/Microsoft.AspNetCore.Session/DistributedSession.cs

```diff
@@ -65,6 +71,33 @@ public class DistributedSession : ISession
             _isNewSessionKey = isNewSessionKey;
         }
 
+        public string Id
+        {
+            get
+            {
+                Load(); // TODO: Silent failure
```

Comment:
I don't get it why we needs strings for logging. 

It looks like retuning a int as a string. There are (almost) no benefits with it. 

All loggers can easily handle objects with can be toStringed. 

IMO the only good reason would be (to have the freedom)  to change in the futher the implementation to a non-guid.